cli/daemon/apps/apps.go

```diff
@@ -0,0 +1,140 @@
+package apps
+
+import (
+	"database/sql"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/rs/zerolog/log"
+
+	"encr.dev/cli/daemon/internal/manifest"
+	"encr.dev/cli/internal/appfile"
+)
+
+var ErrNotFound = errors.New("app not found")
+
+// Instance describes an app instance known by the Encore daemon.
+type Instance struct {
+	root       string
+	localID    string
+	platformID string
+}
+
+func NewInstance(root, localID, platformID string) *Instance {
+	return &Instance{root, localID, platformID}
+}
+
+// Root returns the filesystem path for the app root.
+// It always returns a non-empty string.
+func (i *Instance) Root() string { return i.root }
+
+// LocalID reports a local, random id unique for this app,
+// as persisted in the .encore/manifest.json file.
+// It always returns a non-empty string.
+func (i *Instance) LocalID() string { return i.localID }
+
+// PlatformID reports the Encore Platform's ID for this app.
+// If the app is not linked it reports the empty string.
+func (i *Instance) PlatformID() string { return i.platformID }
```

Comment:
What do you think about having a `type AppID string` type? I'm wondering if it might be nice to add a port of the EID package here down the line - but for now strongly typing the ID to a string at least gives us a easy migration path down the line to make that change.

Reply:
I started with that actually, but we ended up needing `PlatformOrLocalID()` in a lot of places because we have to handle the case where the app is not linked. It felt wrong to consider the LocalID as returning an AppID because it's not really the same thing. Not sure what's best

Reply:
One of the reasones we picked XID was to allow ID's to be generated in a distributed manor without communication - so I think LocalID's and AppID's could be interchangeable - I guess the question is if we need different ID prefixes or not (I'd say yes maybe).

Not sure, hence why I wondered if just having a strong type here is good enough.

Maybe if we want to strongly be able to tell the difference, we should use a stronger `AppOrLocaID` type which then can act basically like concrete `Either[A, B]` type?